tests/Unit/RetryFailedRequestQueueWorkerTest.php

```diff
@@ -0,0 +1,244 @@
+        $failedRequestRepository->shouldReceive("update");
+        $failedRequestRepository->shouldReceive("findOneToRetry")->andReturn($failedRequest, null);
+
+        /** @var MockInterface|WriteApi */
+        $writeApi = Mockery::mock(WriteApi::class);
+        $writeApi->shouldReceive("logisticsOrderPost")->andThrow(new ApiException("ups"));
+
+        /** @var MockInterface */
+        $wordpressMock = Mockery::mock("alias:" . WordPress::class);
+        $wordpressMock->shouldReceive("sendMailToAdmin")->once();
+
+        $worker = new RetryFailedRequestsQueueWorker($failedRequestRepository, $this->logisticsOrderFactory, $writeApi, $this->orderRepository, $this->settingsRepository);
+        $worker->start();
+    }
+
+    /**
+     * We need to isolate this test to able to alias mock the
+     * WordPress class with our helper functions.
+     *
+     * @runInSeparateProcess
+     * @preserveGlobalState disabled
+     */
+    public function test_it_sets_the_order_state_to_fulfillment_error_after_the_third_failed_try()
+    {
+        /** @var FailedRequest|MockInterface */
+        $failedRequest = Mockery::mock(FailedRequest::class);
+        $failedRequest->allows([ "getOrderId" => 4, "setStatus" => null, "incrementFailedAttempts" => null ]);
+        $failedRequest->shouldReceive("getFailedAttempts")->once()->andReturn(3);
+
+        /** @var FailedRequestRepository|MockInterface */
+        $failedRequestRepository = Mockery::mock(FailedRequestRepository::class);
+        $failedRequestRepository->shouldReceive("update");
+        $failedRequestRepository->shouldReceive("findOneToRetry")->andReturn($failedRequest, null);
+
+        /** @var MockInterface|WriteApi */
+        $writeApi = Mockery::mock(WriteApi::class);
+        $writeApi->shouldReceive("logisticsOrderPost")->andThrow(new ApiException("ups"));
+
+        /** @var MockInterface */
+        $wordpressMock = Mockery::mock("alias:" . WordPress::class);
+        $wordpressMock->shouldReceive("sendMailToAdmin");
+
+        /** @var MockInterface|\WC_Order */
+        $order = Mockery::mock("WC_Order");
+        $order->allows([
+            "get_id" => 42
+        ]);
+        $order->shouldReceive("set_status")->once()->andReturn(null);
+        $order->shouldReceive("save")->once();
+
+        /** @var MockInterface|OrderRepository */
+        $orderRepository = Mockery::mock(OrderRepository::class);
+        $orderRepository->allows([
+            "findById" => $order
+        ]);
+
+        $worker = new RetryFailedRequestsQueueWorker($failedRequestRepository, $this->logisticsOrderFactory, $writeApi, $orderRepository, $this->settingsRepository);
+        $worker->start();
+
+        // wc-failed is the value returned by the settings repository
+        $order->shouldHaveReceived("set_status", ["wc-failed"]);
+    }
+
+    public function test_it_ensures_that_the_requests_are_authenticated()
+    {
+        $this->markTestIncomplete();
```

Comment:
there is no test here as far as i understand

Reply:
I have updated the additional info section. I wanted to say that there is one incomplete test regarding authentication, and that will be addressed in a separate PR.